# alsaconf_safe_tmp.patch by Thomas Hood
#
# Create alsaconf.cards temporary file safely
# Always run in modinfo mode
Index: alsa-utils-1.0.18/alsaconf/alsaconf.8
===================================================================
--- alsa-utils-1.0.18.orig/alsaconf/alsaconf.8	2008-10-29 13:42:11.000000000 +0100
+++ alsa-utils-1.0.18/alsaconf/alsaconf.8	2008-11-02 20:59:25.000000000 +0100
@@ -54,9 +54,6 @@
 .B \-l, \-\-legacy
 Check only for legacy non-isapnp cards.
 .TP
-.B \-m, \-\-modinfo
-Read module descriptions instead of reading a card database.
-.TP
 .B \-P, \-\-listprobe
 List the supported legacy card modules.
 .TP
Index: alsa-utils-1.0.18/alsaconf/alsaconf.in
===================================================================
--- alsa-utils-1.0.18.orig/alsaconf/alsaconf.in	2008-11-02 20:59:25.000000000 +0100
+++ alsa-utils-1.0.18/alsaconf/alsaconf.in	2008-11-02 20:59:25.000000000 +0100
@@ -81,7 +81,6 @@
     echo "  version $version"
     xecho "usage: alsaconf [options]
   -l|--legacy    check only legacy non-isapnp cards
-  -m|--modinfo   read module descriptions instead of reading card db
   -s|--sound wav-file
                  use the specified wav file as a test sound
   -u|--uid uid   set the uid for the ALSA devices (default = 0) [obsoleted]
@@ -120,8 +119,9 @@
     case "$1" in
     -l|--legacy)
 	do_legacy_only=1; shift ;;
-    -m|--modinfo)
-	use_modinfo_db=1; shift ;;
+# In Debian the tool always operates in modinfo mode
+#    -m|--modinfo)
+#	use_modinfo_db=1; shift ;;
     -s|--sound)
 	TESTSOUND=$2; shift 2;;
     -h|--help)
@@ -517,10 +517,15 @@
     killall -9 aplay arecord >/dev/null 2>&1
     /sbin/modprobe -r isapnp >/dev/null 2>&1
     /sbin/modprobe -r isa-pnp >/dev/null 2>&1
-    rm -f "$TMP" "$addcfg" "$FOUND" "$DUMP"
+    rm -f "$CARDID_DB" "$TMP" "$addcfg" "$FOUND" "$DUMP"
 }
 trap cleanup 0 
 
+CARDID_DB=`mktemp -q /tmp/alsaconf.cards.XXXXXX`
+if [ $? -ne 0 ]; then
+	xecho "Can't create temp file, exiting..."
+        exit 1
+fi
 TMP=`mktemp -q /tmp/alsaconf.XXXXXX`
 if [ $? -ne 0 ]; then
 	xecho "Can't create temp file, exiting..."
@@ -638,6 +643,16 @@
     if [ "$found" = "0" ]; then
       echo -n >"$DUMP"
     fi 
+if true; then
+    # Debian code
+    # CARDID_DB was set earlier
+    xecho "Building card database..."
+    build_card_db $CARDID_DB
+    if [ ! -e $CARDID_DB ]; then
+        xecho "No card database. Aborting."
+	exit 1
+    fi
+else
     CARDID_DB=/var/tmp/alsaconf.cards
     if [ ! -r $CARDID_DB ]; then
         use_modinfo_db=1
@@ -655,6 +670,7 @@
 	xecho "No card database is found.."
 	exit 1
     fi
+fi
     ncards=`grep '^snd-.*\.o$' $CARDID_DB | wc -w`
 
     msg=$(gettext "Searching sound cards")
